### PR TITLE
Feature/user metrics

### PR DESCRIPTION
**Pull request by cristian-home-codebranch**
## Requirement
[Contract for Usermetrics](https://coda.io/d/IK-Nv-projects_dlsKLnttMd2/MASTER-REQUIREMENTS_suuLGhPJ)

## Description:
Implement user metrics functionality; a request is sent each time the LLM returns a response.

## Summary by CodeRabbit

- **New Features**
  - Chat submits user metrics at stream end (conversationId, query, response, environment, browser, and user/session info); submission is non-blocking.
  - Feedback on AI responses now includes the associated user query for context.

- **Improvements**
  - Improved end-of-stream handling: avoids duplicate assistant messages and better surfaces chart-generation errors or "no response" cases.

- **Configuration / Chores**
  - Added env vars and backend mapping to enable a user-metrics backend.

- **Tests**
  - Expanded test coverage for metrics, validation, environment/browser detection, and streaming UI edge cases.

### CODE REVIEW

File: src/lib/api/utils/validation.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ * @param fields Array of required field names
+ * @returns Missing fields array or null if all fields present
+ */
+export function validateRequiredFields(
+  body: Record<string, unknown>,
+  fields: string[]
+): string[] | null {
+  const missing = fields.filter((field) => !body[field]);
+  return missing.length > 0 ? missing : null;
+}
```
[the remark this answers is not in the document]
Comment (DanielAlferezCode) [reply src/lib/api/utils/validation.ts:53]: Not needed: the current function intentionally treats '', 0, and false as “missing” (non-empty rule).
Tests rely on this behavior; switching to hasOwnProperty would change semantics and break tests.
